### ISSUE TITLE
Compile errors seen on VS2022 after upgrading rapidjson

### ISSUE THREAD

**Issue by mdear** [miloyip/rapidjson#2069]
Hello, 

I'm refreshing a four-year-old project and rebuilding in VS2022, I was having runtime errors with rapidjson so I upgraded to this stream and now am seeing compile errors.  Any idea what may be wrong here ?

<img width="658" alt="VisualStudioRapidJsonCompileErrors_Screenshot 2022-08-31 182610" src="https://user-images.githubusercontent.com/6562082/187796060-8cdced5e-bb4f-478f-8d27-1c38060791ee.png">

**Comment by bostjane** [comment]
Yes, it seems to be wider problem. See my findings in https://github.com/Tencent/rapidjson/issues/2077#issuecomment-1262136344

**Comment by cwe-skov-dk** [comment]
This issue will be fixed by pull request: https://github.com/Tencent/rapidjson/pull/2081
